### PR TITLE
[LOOP-1127] Use TimeInterval type for duration

### PR DESCRIPTION
**Pull request by darinkrauss**
- https://tidepool.atlassian.net/browse/LOOP-1127

### CODE REVIEW

**Review by nhamming — APPROVED** [review]
LGTM

File: TidepoolKit/Model/Datum/BasalDatum/TBasalDatum.swift [hunk 1]
```diff
@@ -33,16 +33,16 @@ public class TBasalDatum: TDatum {
         guard self.deliveryType == expectedDeliveryType else {
             throw DecodingError.dataCorruptedError(forKey: .deliveryType, in: container, debugDescription: "Unexpected deliveryType '\(self.deliveryType)'")
         }
-        self.duration = try container.decodeIfPresent(Int.self, forKey: .duration)
-        self.expectedDuration = try container.decodeIfPresent(Int.self, forKey: .expectedDuration)
+        self.duration = try container.decodeIfPresent(Int.self, forKey: .duration).map { .milliseconds($0) }
```
Comment (rickpasetto) [inline TidepoolKit/Model/Datum/BasalDatum/TBasalDatum.swift:36]: Q: why round to milliseconds?

**Review by rickpasetto — COMMENTED** [review]
Question, @darinkrauss : why round to milliseconds?  Or is that just part of the API definition?

File: TidepoolKit/Model/Datum/BasalDatum/TBasalDatum.swift (hunk 1 above)
Comment (darinkrauss) [reply TidepoolKit/Model/Datum/BasalDatum/TBasalDatum.swift:36]: @rickpasetto The API specification states that the `duration` field (and `expectedDuration`) is a milliseconds-based integer. Therefore, we decode the integer and then convert it to a `TimeInterval` in milliseconds.

Comment (rickpasetto) [reply TidepoolKit/Model/Datum/BasalDatum/TBasalDatum.swift:36]: @darinkrauss Gotcha. 👍

**Review by rickpasetto — APPROVED** [review]
LGTM.

**Pull request merged**